app/controllers/users_controller.rb

```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  include EventsHelper
+  before_action :authenticate_user!
+  def index
+    @users = User.all
+  end
+
+  def new
+    @user = User.current_user.build
+    @past_events = previous_events
+    @upcoming_events = upcoming_events
+  end
+
+  def create
+    @user = User.current_user.build(event_params)
+    @user.save
+    session[:user_id] = @user.id
+    flash.notice = "User #{@user.name} successfully created! Thank you for signing up!"
+    redirect_to user_path(@user)
+  end
+
+  def show
+    @all_events = Event.all
+    @events = Event.where(user_id: current_user.id)
```

Comment:
### MANDATORY

- [ ] Lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [ ] Lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.past` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [ ] Lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

Basically, the user `#show` view should display events that the user has **"created"**, and the events the user is attending, **"separated"** by **"past"** and **"upcoming"**. You can use the query above to get those data